### DISCUSSION TITLE
NU1101: Unable to find package cake.tool

### DISCUSSION THREAD

**Discussion by RehanSaeed** [cake-build/cake#3304] [Q&A] [answered]
I started getting this error today, even though I upgraded to .NET SDK 5.0.202 which supposedly fixes NuGet restore.

```
Run dotnet tool restore
error NU1101: Unable to find package cake.tool. No packages exist with this id in source(s): Microsoft Visual Studio Offline Packages

Package "cake.tool" failed to restore, due to Microsoft.DotNet.ToolPackage.ToolPackageException: The tool package could not be restored.
   at Microsoft.DotNet.Tools.Tool.Install.ProjectRestorer.Restore(FilePath project, PackageLocation packageLocation, String verbosity)
   at Microsoft.DotNet.ToolPackage.ToolPackageInstaller.InstallPackageToExternalManagedLocation(PackageLocation packageLocation, PackageId packageId, VersionRange versionRange, String targetFramework, String verbosity)
   at Microsoft.DotNet.Tools.Tool.Restore.ToolRestoreCommand.InstallPackages(ToolManifestPackage package, Nullable`1 configFile)

Restore failed.
Error: Process completed with exit code 1.
```
https://github.com/Dotnet-Boxed/Templates/runs/2286810853?check_suite_focus=true

**Comment by augustoproiete** [comment, marked as answer]
Looks like a bug in the restore process that decided not to check nuget.org and instead is only looking at the offline packages source.

You might want to try to drop a `nuget.config` file in the root of the repo to see if that forces nuget.org to be searched:

```xml
<?xml version="1.0" encoding="utf-8"?>
<configuration>
  <packageSources>
    <add key="nuget.org" value="https://api.nuget.org/v3/index.json" protocolVersion="3" />
  </packageSources>
</configuration>
```

**Reply by gep13** [reply]
Seems like this is not an isolated issue, it is happening in multiple places: https://github.com/actions/virtual-environments/issues/1090#issuecomment-814783299

**Reply by devlead** [reply]
I've been having issues for months, my remedy has been

- nuget.config in repo
- clear nuget cache using .NET CLI

**Reply by RehanSaeed** [reply]
Added a `nuget.config` and that fixed it. Ridiculous. I'd seen people complaining on Twitter for weeks but had not got stung until now and even then only on one repo. Thanks for the link.

**Comment by augustoproiete** [comment]
Another possible workaround, which can be added as one of the job steps:

```powershell
dotnet nuget add source https://api.nuget.org/v3/index.json -n nuget.org --configfile $env:APPDATA\NuGet\NuGet.Config
```
